geonode_mapstore_client/client/js/components/DetailsPanel/DetailsInfo.jsx

```diff
@@ -156,7 +163,8 @@ function DetailsInfo({
                 items: isDefaultTabType(tab.type) ? parseTabItems(tab?.items) : tab?.items,
                 Component: tabTypes[tab.type] || tabTypes.tab
             }))
-        .filter(tab => tab?.items?.length > 0 );
+        // ensure tab has items .. attribute table loads them dynamically
+        .filter(tab => tab?.items?.length > 0 || canShowAttributeTab({resource, tab}) );
```

Comment:
I would propose also to use [expression](https://docs.mapstore.geosolutionsgroup.com/en/latest/developer-guide/plugins-documentation/#dynamic-configuration) inside the localConfig.json to get the property directly from the loaded resources, something like this:

```
{
    "type": "attribute-table",
    "id": "attributes",
    "labelId": "gnviewer.attributes",
    "disableIf": "{context.get(state('gnResourceData'), 'resource_type') !== 'dataset'}",
    "items": "{context.get(state('gnResourceData'), 'attribute_set')}"
},
```

GeoNode mapstore client is [monitoring some part of the state](https://github.com/GeoNode/geonode-mapstore-client/blob/master/geonode_mapstore_client/static/mapstore/configs/localConfig.json#L38-L75) including the current selected resource (gnResourceData) in addition is [exposing some function](https://github.com/GeoNode/geonode-mapstore-client/blob/master/geonode_mapstore_client/client/js/utils/PluginsContextUtils.js) that could be used inside the expression using the context object.
The reason of this proposal is to avoid specific checks such as `canShowAttributeTab`.
Note I added also the disebleIf property that could be use to completely filter the tab out before proceeding into a validation process

Reply:
Also good point .. that makes it overall more consistent